### ISSUE TITLE
Incorrect warning about a missing fallback image

### ISSUE THREAD

**Issue by cecchi** [imgix/react-imgix#164]
**Before you submit:**

- [x] Please search through the existing issues (both open AND closed) to see if your issue has been discussed before. Github issue search can be used for this: https://github.com/imgix/react-imgix/issues?utf8=%E2%9C%93&q=is%3Aissue
- [x] Please ensure the problem has been isolated and reduced. This link explains more: http://css-tricks.com/6263-reduced-test-cases/

**Describe the bug**
The library warns about a missing fallback `<img />` or `<Imgix />` tag when those components not immediate descendants. 

We have wrapped `<Image />`, `<Source />` and `<Picture />` with our own small wrapper components to set defaults and tweak the behavior slightly. The library iterates through `this.props.children` and explicitly looks for the ReactImgix constructors or a ReactDOM "img" component. `react-imgix` should provide an escape hatch to prevent this warning (some kind of static decoration on the wrapper component, maybe), or should recursively iterate through the children to find the fallback component.

**To Reproduce**
```tsx
import Imgix, {
  Source as ImgixSource,
  Picture as ImgixPicture
} from 'react-imgix'

// Set some default URL params
function addDefaultImgixParams(params: ImgixProps['imgixParams']) {
  return Object.assign(
    {
      auto: ['compress', 'format'], // Default to compressing and reformatting (https://docs.imgix.com/apis/url/auto)
      ch: ['Width', 'DPR', 'Save-Data'] // Default to setting all client-hints (https://docs.imgix.com/apis/url/format/ch)
    },
    params
  )
}

// Wrap <Image />
export interface ImageProps extends ImgixProps {}
const Image: React.SFC<ImageProps> = ({ imgixParams, ...props }) => {
  imgixParams = addDefaultImgixParams(imgixParams)

  return <Imgix imgixParams={imgixParams} {...props} />
}

// Wrap <Source />
export interface SourceProps extends ImgixSourceProps {}
const Source: React.SFC<SourceProps> = ({ imgixParams, ...props }) => {
  imgixParams = addDefaultImgixParams(imgixParams)

  return <ImgixSource imgixParams={imgixParams} {...props} />
}

// Wrap <Picture />
export interface PictureProps extends ImgixPictureProps {}
const Picture: React.SFC<PictureProps> = props => <ImgixPicture {...props} />

// Attempt to render a responsive image:
<Picture>
  <Source ... />
  <Image ... />
</Picture>

// ^^ This results in a warning.
```

**Expected behaviour**
The library should not log a warning to the console if the resulting markup includes a valid fallback image.

**Information:**

- react-imgix version: v8.0.0

**Additional context:**

This is a useful warning, but unfortunately is implemented in such a way that isn't flexible enough.

**Comment by frederickfogerty** [comment]
Hi there @cecchi, thanks for the bug report. I understand that this is frustrating, so sorry about that! 

> should recursively iterate through the children to find the fallback component.

I'm liking the sound of this approach. If you have any time to submit a PR, I'll happy review it! Otherwise, I've added this to our backlog, and I'll hope to have a look at it in the next few weeks.

**Comment by cecchi** [comment]
With some more thought, recursively iterating through children isn't a robust or feasible solution. The children could be using other rendering patterns like render props that would make it impossible to determine whether it will result in "valid" markup. Additionally, `react-imgix` will mutate the `children` prop to ensure the `<img>` is the final child... this will not be possible with more complex children.

In my opinion, the warning should be removed, and the logic for mutating children should only apply if all children are `<Source />` or `<Image />`. In general, mutating children seems unexpected, so that case may be even better off as a warning.

**Comment by frederickfogerty** [comment]
This was fixed by #168
